hbase-common/src/main/java/org/apache/hadoop/hbase/util/CommonFSUtils.java

```diff
@@ -350,11 +349,19 @@ public static void setWALRootDir(final Configuration c, final Path root) {
   public static FileSystem getWALFileSystem(final Configuration c) throws IOException {
     Path p = getWALRootDir(c);
     FileSystem fs = p.getFileSystem(c);
-    // hadoop-core does fs caching, so need to propogate this if set
+    // hadoop-core does fs caching, so need to propagate this if set
     String enforceStreamCapability = c.get(UNSAFE_STREAM_CAPABILITY_ENFORCE);
     if (enforceStreamCapability != null) {
       fs.getConf().set(UNSAFE_STREAM_CAPABILITY_ENFORCE, enforceStreamCapability);
     }
+    if (!c.getBoolean(HConstants.CLUSTER_DISTRIBUTED, false)
```

Comment:
Just test whether this is a LocalFileSystem is enough?

Reply:
You tell me :)

Yes, for running out of the source tree, not modifying any configuration, this change permits the process to start, master and region server threads all come online. I performed some additional verification by running `ltt` against the process, a scant 100k rows.

I don't know well when we choose `LocalFileSystem` vs. `RawLocalFIleSystem`, so there may be other issues lurking.

Reply:
Hmm.. I too think it should just be (fs instanceof LocalFS).. With this patch, try using the following combination (distributed cluster + localfs), it fails with the same error when it should work..

```
<configuration>
    <property>
          <name>hbase.cluster.distributed</name>
          <value>true</value>
    </property>
</configuration>
```

Point being that, these two configuration params are disjoint. Meaning a distributed/non-distributed cluster should work with a local file system. Stream capability is a property of the FS in use, so just a check on FS should be good IMO.

I poked around a little bit about RawFS in the hadoop code, didn't fully understand its use but it appears that FileSystem when parsing the config would pick LocalFileSystem when the scheme is file:///. So I think just doing it for LocalFileSystem should be ok.

Reply:
How do you run a distributed cluster on local fs? You running off of shared NFS mounts?

Reply:
Pseudo-distributed http://hbase.apache.org/book.html#quickstart_pseudo, just that I didn't set "hbase.rootdir" and hence it defaults to local fs.

Reply:
Yeah, fair enough. So I'll remove the `CLUSTER_DISTRIBUTED` check and instead check `fs instanceOf LocalFileSystem`.

I also asked the stream capabilities question over on [hdfs-user](https://lists.apache.org/thread.html/r890ad8e0c25be353ab400b677ed3c0387e12b987e4826b0748e9735a%40%3Chdfs-user.hadoop.apache.org%3E).